crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -0,0 +1,248 @@
+    replacement: String,
+}
+
+impl AlwaysFixableViolation for UnnecessaryRegularExpression {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Plain string pattern passed to `re` function".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        format!("Replace with `{}`", self.replacement)
+    }
+}
+
+/// RUF055
+pub(crate) fn unnecessary_regular_expression(checker: &mut Checker, call: &ExprCall) {
+    // adapted from unraw_re_pattern
+    let semantic = checker.semantic();
+
+    if !semantic.seen_module(Modules::RE) {
+        return;
+    }
+
+    let Some(qualified_name) = semantic.resolve_qualified_name(&call.func) else {
+        return;
+    };
+
+    let ["re", func] = qualified_name.segments() else {
+        return;
+    };
+
+    // skip calls with more than `pattern` and `string` arguments (and `repl`
+    // for `sub`)
+    let Some(re_func) = ReFunc::from_call_expr(semantic, call, func) else {
+        return;
+    };
+
+    // For now, restrict this rule to string literals
+    let Some(string_lit) = re_func.pattern.as_string_literal_expr() else {
+        return;
+    };
+
+    // For now, reject any regex metacharacters. Compare to the complete list
+    // from https://docs.python.org/3/howto/regex.html#matching-characters
+    let has_metacharacters = string_lit.value.to_str().contains([
+        '.', '^', '$', '*', '+', '?', '{', '}', '[', ']', '\\', '|', '(', ')',
+    ]);
+
+    if has_metacharacters {
+        return;
+    }
+
+    // Here we know the pattern is a string literal with no metacharacters, so
+    // we can proceed with the str method replacement
+    let new_expr = re_func.replacement();
+
+    let repl = checker.generator().expr(&new_expr);
+    let mut diagnostic = Diagnostic::new(
+        UnnecessaryRegularExpression {
+            replacement: repl.clone(),
+        },
+        call.range,
+    );
+
+    let edit = Edit::range_replacement(repl, call.range);
+    let fix = if checker
+        .comment_ranges()
+        .has_comments(call, checker.source())
+    {
+        Fix::unsafe_edit(edit)
+    } else {
+        Fix::safe_edit(edit)
+    };
+
+    diagnostic.set_fix(fix);
+
+    checker.diagnostics.push(diagnostic);
+}
+
+/// The `re` functions supported by this rule.
+#[derive(Debug)]
+enum ReFuncKind<'a> {
+    Sub { repl: &'a Expr },
+    Match,
+    Search,
+    Fullmatch,
+    Split,
+}
+
+#[derive(Debug)]
+struct ReFunc<'a> {
+    kind: ReFuncKind<'a>,
+    pattern: &'a Expr,
+    string: &'a Expr,
+}
+
+impl<'a> ReFunc<'a> {
+    fn from_call_expr(
+        semantic: &SemanticModel,
+        call: &'a ExprCall,
+        func_name: &str,
+    ) -> Option<Self> {
+        let nargs = call.arguments.len();
+
+        // the proposed fixes for match, search, and fullmatch rely on the
+        // return value only being used for its truth value
+        let in_if_context = semantic.in_boolean_test();
+
+        match (func_name, nargs) {
```

Comment:
nit: I'd just inline `nargs`

```suggestion
        // the proposed fixes for match, search, and fullmatch rely on the
        // return value only being used for its truth value
        let in_if_context = semantic.in_boolean_test();

        match (func_name, call.arguments.len()) {
```